gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineVersions.java

```diff
@@ -53,19 +60,33 @@
 
     @Override
     public void apply(Project project) {
-        File rootVersionsPropsFile = rootVersionsPropsFile(project);
 
         // apply plugin: "nebula.dependency-recommender"
         project.getPluginManager().apply(DependencyRecommendationsPlugin.class);
 
         // get dependencyRecommendations extension
-        RecommendationProviderContainer extension = project.getExtensions()
+        RecommendationProviderContainer extension = project
+                .getExtensions()
                 .getByType(RecommendationProviderContainer.class);
 
-        extension.setStrategy(RecommendationStrategies.OverrideTransitives); // default is 'ConflictResolved'
+        extension.setStrategy(RecommendationStrategies.OverrideTransitives); // default is 'ConflictResolved';
+        extension.whenObjectAdded(recommendationProvider -> applyPropsFile(project));
```

Comment:
this is actually a bit more annoying since if you have multiple boms it's still not what you expect - happy to outlaw multiple boms anyway

Reply:
Actually - bom recommendations are always first so no need to do this actually

Reply:
Ok. I still need to make sure the mavenBom configuration have been applied so I probably want to make my task check task depend on "resolveConfiguration". 
Could you point me to the logic where you found that the mavenBom recommendations are always applied first ? I remember looking at it and it was a bit unclear if it was or not.

Reply:
Don't outlaw multiple boms! It's been useful a number of times for me

Reply:
this logic https://github.com/nebula-plugins/nebula-dependency-recommender-plugin/blob/master/src/main/groovy/netflix/nebula/dependency/recommender/provider/RecommendationProviderContainer.java#L59 - the provider is always added upon creation of extension and subsequent invocations are just adding to the NEBULA_RECOMMENDER_BOM configuration

Reply:
I see. I remember not understanding at first read why they treated the mavenbomrecommendation provider differently than the others (as in, not just adding it as yet another recommendation provider) but that must be it. Thanks!